src/mmds/src/data_store.rs

```diff
@@ -112,21 +123,53 @@ impl Mmds {
     }
 
     /// Set the MMDS version.
-    pub fn set_version(&mut self, version: MmdsVersion) {
-        self.version = version;
+    pub fn set_version(&mut self, version: MmdsVersion) -> Result<(), Error> {
+        match version {
+            MmdsVersion::V1 => {
+                self.token_authority = None;
+                Ok(())
+            }
+            MmdsVersion::V2 => {
+                if self.token_authority.is_none() {
+                    self.token_authority =
+                        Some(TokenAuthority::new().map_err(Error::TokenAuthority)?);
+                }
```

Comment:
Can we do this with `Option::or`?

Reply:
We can’t use `Option::or` to replace here. `Token_authority` will be moved because it does not derive the `Copy` trait. This is due to the fact that `TokenAuthority` struct contains a `File` field, which also does not derive `Copy`. I propose to leave this as is.